server/src/main/java/com/cloud/configuration/Config.java

```diff
@@ -265,7 +265,7 @@ public enum Config {
     NetworkRouterRpFilter(
             "Network",
             ManagementServer.class,
-            Integer.class,
+            Boolean.class,
```

Comment:
Upgrading any existing env with current int value should change value to true / false.

Reply:
Env with an int value should change to false.

Reply:
@nvazquez Agree that both are enabled when the value is 'true', and also, the default value set is 'true'. As the config is set to integer type, the update config fails to take the boolean values, and only accepts integer values. If anyone have update the config with integer value, then it is disabled, so upgrade path should set to 'false' accordingly. What is your opinion?

```
(test-env) 🐱 > list configurations name=rpfilter filter=name,value
{
  "configuration": [
    {
      "name": "consoleproxy.disable.rpfilter",
      "value": "true"
    },
    {
      "name": "network.disable.rpfilter",
      "value": "true"
    }
  ],
  "count": 2
}
(test-env) 🐱 > update configuration name=network.disable.rpfilter value=false
🙈 Error: (HTTP 431, error code 4350) There was error in trying to parse value: false. Please enter a valid integer value for parameter network.disable.rpfilter
(test-env) 🐱 > 
(test-env) 🐱 > update configuration name=network.disable.rpfilter value=3
{
  "configuration": {
    "category": "Network",
    "description": "disable rp_filter on Domain Router VM public interfaces.",
    "isdynamic": false,
    "name": "network.disable.rpfilter",
    "value": "3"
  }
}
(test-env) 🐱 > 
(test-env) 🐱 > update configuration name=consoleproxy.disable.rpfilter value=true
🙈 Error: (HTTP 431, error code 4350) There was error in trying to parse value: true. Please enter a valid integer value for parameter consoleproxy.disable.rpfilter
(test-env) 🐱 > 
(test-env) 🐱 > update configuration name=consoleproxy.disable.rpfilter value=false
🙈 Error: (HTTP 431, error code 4350) There was error in trying to parse value: false. Please enter a valid integer value for parameter consoleproxy.disable.rpfilter
(test-env) 🐱 > 
(test-env) 🐱 > update configuration name=consoleproxy.disable.rpfilter value=9
{
  "configuration": {
    "category": "Console Proxy",
    "description": "disable rp_filter on console proxy VM public interface",
    "isdynamic": false,
    "name": "consoleproxy.disable.rpfilter",
    "value": "9"
  }
}
(test-env) 🐱 > 
(test-env) 🐱 > 
(test-env) 🐱 > list configurations name=rpfilter filter=name,value
{
  "configuration": [
    {
      "name": "consoleproxy.disable.rpfilter",
      "value": "9"
    },
    {
      "name": "network.disable.rpfilter",
      "value": "3"
    }
  ],
  "count": 2
}
(test-env) 🐱 > 
```

Reply:
@sureshanaparti I'm totally agree with you. I don't know how I can achieve that goal. Do you have any suggestion?

Reply:
@soreana Try updating the config value to 'false' in the upgrade path (either Upgrade class / schema file) when the value is not 'true'.